Makefile

```diff
@@ -1,41 +1,61 @@
-TEST_CMAKE_BUILD_DIR=cmake-build
-TEST_CMAKE_INSTALL_DIR=cmake-install
-TEST_CMAKE_SCRATCH_DIR=cmake-scratch
-
+ROOT:=$(shell pwd)
+TEST_DIR=$(ROOT)/.test-output
 
 .PHONY: test
 test:
 
 
 .PHONY: test_cmake
 test: test_cmake
-test_cmake: $(TEST_CMAKE_SCRATCH_DIR)
+test_cmake: $(TEST_DIR)/cmake-scratch
+
+
+.PHONY: test_cmake_add_subdir
+test: test_cmake_add_subdir
+test_cmake_add_subdir: $(TEST_DIR)/ciabatta-copy
+
 
+$(TEST_DIR)/cmake-build: CMakeLists.txt test/CMakeLists.txt
+	rm -rf $(TEST_DIR)/cmake-build
+	mkdir -p $(TEST_DIR)/cmake-build
+	cd $(TEST_DIR)/cmake-build && \
```

Comment:
wait... this test will fail if the user doesn't have `ninja` locally installed, right? `ninja` doesn't come out of the box with Windows/macOS. Should this use MSVC on Windows and Xcode on macOS instead? (and make on Linux)?

Reply:
I guess it's possible to use cmake to invoke the driver, I just haven't figured how yet (should be easy though).
This is automating the testing for deployment, *with a makefile* :). I'm not even sure how one runs it on windows.

Would you agree that generalizing the testing infra is a separate PR?

Reply:
Sure, that's a separate issue. Feel free to merge this then.